sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java

```diff
@@ -890,7 +892,11 @@ HttpPipeline setupPipeline(HttpClient httpClient) {
         policies.addAll(options.getPerCallPolicies());
         HttpPolicyProviders.addBeforeRetryPolicies(policies);
         // Add retry policy.
-        policies.add(ClientBuilderUtil.validateAndGetRetryPolicy(options.getRetryPolicy(), options.getRetryOptions()));
+        RetryPolicy retryPolicy = options.getRetryPolicy();
+        if (retryPolicy == null && options.getUseImdsRetryStrategy()) {
+            retryPolicy = new RetryPolicy(new ImdsRetryStrategy(5));
```

Comment:
Can we either have a parameterless ctor for `ImdsRetryStrategy` that defaults to 5, or make this a constant with a comment somewhere so we know why it is 5?